app/services/audit/catalog_to_moab.rb

```diff
@@ -3,43 +3,30 @@
 module Audit
   # Catalog to Moab existence check code
   class CatalogToMoab
-    attr_reader :complete_moab, :druid, :logger
+    attr_reader :complete_moab, :logger
 
     def initialize(complete_moab)
       @complete_moab = complete_moab
-      @druid = complete_moab.preserved_object.druid
       @logger = Logger.new(Rails.root.join('log', 'c2m.log'))
     end
 
-    # shameless green implementation
+    # Check the catalog version (complete moab version) against versions of the preserved object and the moab on storage
     def check_catalog_version
```

Comment:
I don't have specific qualms with any of the changes in this class, but in total, the class is iceberg-like: it's bottom-heavy with the public API doing little other than invoking private methods*. We are guilty of this in lots of places, I concede.

* I do like the intention-revealing names of these methods! I'm more wondering if we shouldn't inline more of the code lurking in the private methods.

Reply:
Yes, this is quite deliberate and I've been using this strategy throughout prescat. The goal is to provide a grokkable version of what actions the class is doing. This is particularly important because many of the prescat classes involve many steps / conditionals. With the previous inline code you'd lose track of what was happening before you got through the main method.

This could also be accomplished with comments, but comments always end up out of sync.

Open to suggestions about how to adjust this strategy and/or specific tweaks.

Reply:
OK, then let's roll with it. 😄